app_test.py

```diff
@@ -121,4 +121,5 @@ def user_input_images(query):
 if __name__ == "__main__":
     if query_text_input != '':
         user_input(query_text_input)
```

Comment:
High severity: Hard-coded API key detected, exposing credentials in source code is a critical security risk, score: 5/10. Use environment variables or a secure secrets manager to store keys and rotate compromised keys. Suggest this fix:

```suggestion
import os
client = genai2.Client(api_key=os.getenv("GENAI_API_KEY"))
```

Also, search the repo for other committed secrets and rotate any that were exposed.